### ISSUE TITLE
Tox 0.2.9 silently skips tests

### ISSUE THREAD

**Issue by xWNgLWT0mQf** [TokTok/c-toxcore#1308]
When trying to build toxcore 0.2.9, tests are silently skipped. The flags that I have set are in flags.pdf and the build log is in build.log (obviously). Note lines 125 to 130 in build.log for the problem. I wrote a Gentoo installer script (ebuild) and an installing from that. This script calls the standard build script though. If you need more details, let me know.

[Flags.pdf](https://github.com/TokTok/c-toxcore/files/2851009/Flags.pdf)
[build.log](https://github.com/TokTok/c-toxcore/files/2851010/build.log)

**Comment by zugz** [comment]
Could you provide the ebuild?

My first guess is that you need to enable the AUTOTEST flag, which was 
recently introduced, defaults to OFF, and controls whether the 
integration tests are built.

**Comment by xWNgLWT0mQf** [comment]
That fixed it! Out of curiosity, why is this flag there? If it is required to run tests, why offer the user the choice of disabling it? As requested though, I have included the install script. The file tox-0.2.9_ebuild.txt should be called tox-0.2.9.ebuild, but Github blocks that upload. I have also included the successful build log.
[tox_0_2_9_updated_build.log](https://github.com/TokTok/c-toxcore/files/2858274/tox_0_2_9_updated_build.log)
[tox-0.2.9_ebuild.txt](https://github.com/TokTok/c-toxcore/files/2858280/tox-0.2.9_ebuild.txt)

**Comment by nurupo** [comment]
>Out of curiosity, why is this flag there?

Well, we have `STRICT_ABI` option for some time now, it's mostly used by package maintainers to build toxcore package with only API symbols being exported in the ABI. However, after Debian package maintainer complained, we discovered that the option was actually broken for a few releases now, it was conflicting with building autotests, as they need to call internal toxcore functions, they test those. So we added an option to enable autotests, which is set to "off" when `STRICT_ABI` is specified. Not sure why it was made to be set to "off" by default though.
